woocommerce/woocommerce-gutenberg-products-block
# Create withProducts HOC for All Products block

Fixes #977.

Based on #899.

This PR:
* Hooks up the _All Products_ block to the API via a `withProducts` HOC.
* Adds a placeholder state to components so they don't break while loading.
* Adds an _Order by_ select.
* Adds pagination to _All Products_.

### Screenshots

![Peek 2019-09-26 15-23](https://user-images.githubusercontent.com/3616980/65691818-d0c4de80-e071-11e9-8404-4131796e4933.gif)

### How to test the changes in this Pull Request:

Note: I needed to comment out [these lines](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/add/all-products-block/src/RestApi.php#L53-L56) to get API calls to work properly.

Note 2: Currently, the `products` endpoint only works for users with permissions. So you need to be logged in to get this block to work.

1. Create a _All Products_ block.
2. Verify it shows placeholder while loading.
3. Verify it loads data from the API.
4. Verify you can change the order and navigate the different pages.
5. In the editor, test the _List Settings_ attributes: hide the order by select and change the default order.
